tests/api.v2/model/experiment.test.js

```diff
@@ -1,30 +1,121 @@
+}));
+
+jest.mock('../../../src/sql/helpers', () => ({
+  aggregateIntoJson: jest.fn(),
+}));
+
 const experiment = require('../../../src/api.v2/model/experiment');
 
+const mockExperimentId = 'mockExperimentId';
+
 describe('model/experiment', () => {
-  it('Returns the correct generateBasicModelFunctions', async () => {
-    expect(generateBasicModelFunctions).toHaveBeenCalledTimes(1);
-    expect(generateBasicModelFunctions).toHaveBeenCalledWith({
-      tableName: 'experiment',
-      selectableProps: [
-        'id',
-        'name',
-        'description',
-        'processing_config',
-        'notify_by_email',
-        'created_at',
-        'updated_at',
-      ],
-    });
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
 
+  it('Returns the correct generateBasicModelFunctions', async () => {
     expect(experiment).toEqual(
       expect.objectContaining({
         hasFakeBasicModelFunctions: true,
       }),
     );
   });
+
+  it('getExperimentData works correctly', async () => {
+    const mockAggregateIntoJsonResult = ['result'];
+    helpers.aggregateIntoJson.mockReturnValueOnce(mockAggregateIntoJsonResult);
+
+    const result = await experiment.getExperimentData(mockExperimentId);
+
+    expect(result).toEqual(mockAggregateIntoJsonResult[0]);
+
+    expect(sqlClient.get).toHaveBeenCalled();
+    expect(helpers.aggregateIntoJson).toHaveBeenCalledWith(
+      expect.anything(),
+      ['id', 'name', 'description', 'samples_order', 'notify_by_email', 'created_at', 'updated_at'],
+      ['params_hash', 'state_machine_arn', 'execution_arn'],
+      'pipeline_type',
+      'pipelines',
+      mockSqlClient,
+    );
+
+    const firstParam = helpers.aggregateIntoJson.mock.calls[0][0];
+
+
+    // The replace(/__cov) is to remove coverage annotations:
+    // https://stackoverflow.com/questions/30470796/function-equality-assertion-broken-by-code-coverage-report
+    const firstParamWithoutCoverageAnnotations = firstParam.toString().replace(/cov.*?;/g, '');
+
+    // Checking query (the first param that was passed to helpers.aggregateIntoJson)
+    expect(firstParamWithoutCoverageAnnotations).toMatchSnapshot();
+  });
+
+  it('updateSamplePosition works correctly if valid params are passed', async () => {
+    mockTrx.returning.mockImplementationOnce(
+      () => Promise.resolve([{ samples_order: validSamplesOrderResult }]),
+    );
+    mockTrx.raw.mockImplementationOnce(() => 'resultOfSql.raw');
+
+    await experiment.updateSamplePosition(mockExperimentId, 0, 1);
+
+    expect(mockTrx).toHaveBeenCalledWith('experiment');
+
+    expect(mockTrx.update).toHaveBeenCalledWith({ samples_order: 'resultOfSql.raw' });
+    expect(mockTrx.raw.mock.calls[0]).toMatchSnapshot();
+    expect(mockTrx.where).toHaveBeenCalledWith('id', 'mockExperimentId');
+    expect(mockTrx.returning).toHaveBeenCalledWith(['samples_order']);
+
+    expect(mockTrx.commit).toHaveBeenCalled();
+    expect(mockTrx.rollback).not.toHaveBeenCalled();
```

Comment:
Is there an easy way to check the calls to these functions instead of doing them one by one?

Reply:
Talked over in slack, we looked at some options and decided to leave it to another ticket